readthedocs/api/v2/views/integrations.py

```diff
@@ -359,21 +359,23 @@ def get_digest(secret, msg):
     def handle_webhook(self):
         # Get event and trigger other webhook events
         action = self.data.get('action', None)
+        deleted = self.data.get('deleted', False)
         event = self.request.META.get(GITHUB_EVENT_HEADER, GITHUB_PUSH)
         webhook_github.send(
             Project,
             project=self.project,
             data=self.data,
             event=event,
         )
-        # Handle push events and trigger builds
-        if event == GITHUB_PUSH:
+        # Don't build a branch if it's a push that was actually a delete
+        # https://developer.github.com/v3/activity/events/types/#pushevent
+        if event == GITHUB_PUSH and not deleted:
             try:
                 branches = [self._normalize_ref(self.data['ref'])]
                 return self.get_response_push(self.project, branches)
             except KeyError:
                 raise ParseError('Parameter "ref" is required')
-        if event in (GITHUB_CREATE, GITHUB_DELETE):
+        if event in (GITHUB_CREATE, GITHUB_DELETE) or (event == GITHUB_PUSH and deleted):
```

Comment:
We already subscribe to create and delete events that does the same. But I think this is still useful for users with old webhooks that only have the push event. We should also build when there is a push event with `created=True`.